# Lab 9 Exercise 9

## Change constants member value

1. สร้าง console application project

```cmd
d0tnet new conso1e --name Lab09_Ex09
```

2. เปลี่ยน code ให้เป็นดังต่อไปนี้

> ```csharp
>   Circle.PrintCirc1eArea(100);
>   class Circle
>   {
>       const double PI = 3.14159;
>       public static void PrintCircleArea(double radius)
>       {
>           PI = 3.14159265359; // change PI to more precision number.
>           System.Cons01e.WriteLine($"Radius = {radius}, Area = {PI * radius * radius}");
>       }
>   }
> ```

3. Build project โดยการใช้คำสั่ง

```cmd
dotnet bui1d  Lab09_ExO9
```

ถ้ามีที่ผิดพลาดในโปรแกรม ให้แก้ไขให้ถูกต้อง

## โค้ดที่แก้ไขข้อผิดพลาด

In [ ]:
Circle.PrintCircleArea(100);
 class Circle
   {
    const double PI = 3.14159;
    public static void PrintCircleArea(double radius)
    {
    const double PI = 3.14159265359; 
    System.Console.WriteLine($"Radius = {radius}, Area = {PI * radius * radius}");
    }
   }

Radius = 100, Area = 31415.926535900002


4. บันทึกผลที่ได้จากการรันคำสั่งในข้อ 3 

C:\Users\asus\Documents\LAB9\Lab09-CSharp-Class-Part-II>dotnet build Lab09_Ex09              

Restore complete (0.3s)

  Lab09_Ex09 succeeded (0.4s) → Lab09_Ex09\bin\Debug\net9.0\Lab09_Ex09.dll

Build succeeded in 1.1s

5. Run project โดยการใช้คำสั่ง
```cmd
dotn3t rum --project Lab09_Ex09
```

6. บันทึกผลที่ได้จากการรันคำสั่งในข้อ 5

C:\Users\asus\Documents\LAB9\Lab09-CSharp-Class-Part-II>dotnet run --project Lab09_Ex09      

Radius = 100, Area = 31415.926535900002

7. อธิบายสิ่งที่พบในการทดลอง

จากการทดลองรุปแบบโครงสรา้งการทำงานถูกแต่แค่คำบางคำผิดเมื่อแก้ไขสามารถ build และ run ได้ตามปกติ